back/engines/commercial/verification/app/controllers/verification/patches/omniauth_callback_controller.rb

```diff
@@ -32,8 +32,9 @@ def verification_callback(verification_method)
               redirect_to url
             rescue VerificationService::VerificationTakenError
               fail_verification('taken')
-            rescue VerificationService::NotEntitledError
-              fail_verification('not_entitled')
+            rescue VerificationService::NotEntitledError => e
+              message = e.why ? "not_entitled_#{e.why}" : 'not_entitled'
```

Comment:
Aren't all (or many) `NotEntitledError`s have `why`? E.g., these ones do https://github.com/CitizenLabDotCo/citizenlab/blob/4279dc5e4dbff11b841f8840d8581622afbc9bd6/back/engines/commercial/id_gent_rrn/app/lib/id_gent_rrn/gent_rrn_verification.rb#L74-L75

It means a different new message will be returned for them. But the corresponding copies do not exist.

Reply:
It seems `why` was never used before. So, maybe we could leave it blank in these two (and maybe some other) examples. And so, using this code `message = e.why ? "not_entitled_#{e.why}" : 'not_entitled'` will not change the behaviour for other errors.

Reply:
See [this comment](https://github.com/CitizenLabDotCo/citizenlab/pull/5875#discussion_r1341507955)

Reply:
Also, how could my changes in the `omniauth_callback-controller` affect error messages from gent verification (which doesn't use omniauth)?

Reply:
> Also, how could my changes in the omniauth_callback-controller affect error messages from Gent verification (which doesn't use omniauth)?

I think you're right.